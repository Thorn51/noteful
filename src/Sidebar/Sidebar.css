.folder {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 75px;
  margin: 10px;
  border: lightgray solid 1px;
}

.folder_name {
  font-size: 1.5rem;
}
