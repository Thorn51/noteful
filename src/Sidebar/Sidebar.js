import React from "react";

export default function Sidebar(props) {
  const folderList = Object.keys(props.folders).map((key, index) => {
    return (
      <div key={index}>
        <h2 className="folder_name">{props.folders[key].name}</h2>
      </div>
    );
  });
  return <section className="folder_list">{folderList}</section>;
}
