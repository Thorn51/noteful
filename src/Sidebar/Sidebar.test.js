import React from "react";
import ReactDOM from "react-dom";
import Sidebar from "./Sidebar";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("Sidebar", () => {
  it("renders without throwing errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Sidebar />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it("renders the component as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Sidebar />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
