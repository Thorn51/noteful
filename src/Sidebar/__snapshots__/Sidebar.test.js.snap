// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders the component as expected 1`] = `
<aside
  className="folder_sidebar"
>
  <section
    className="folder_list"
  />
  <a
    href="/add-folder"
    onClick={[Function]}
  >
    <div
      className="add_folder"
    >
      Add Folder
    </div>
  </a>
</aside>
`;
